code/modules/mob/living/carbon/xenomorph/mutators/strains/carrier/shaman.dm

```diff
@@ -354,89 +298,113 @@
 
 	if(!do_after(src, action_def.windup_delay, INTERRUPT_ALL, ACTION_BLUE_POWER_UP))
 		BD.used_shaman_ability = FALSE
-		return
+		return FALSE
 
-	plasma_stored -= 300
+	plasma_stored -= 250
 	eggs_cur--
 
-	var/effect_power = 0
-
 	var/image/effect_overlay = get_busy_icon(ACTION_BLUE_POWER_UP)
 
-	visible_message(SPAN_XENONOTICE("\The [src] squashes the egg into a pulp."), SPAN_XENONOTICE("You squash the egg into a mess of acid, blood, and gore! You absorb the adrenal pheremones and redirect them outwards!"), null, 8)
+	visible_message(SPAN_XENONOTICE("\The [src] squashes the egg into a pulp."), SPAN_XENONOTICE("You squash the egg into a mess of acid, blood, and gore! You absorb the pheremones and redirect them outwards!"), null, 8)
 
-	for(var/mob/living/carbon/Xenomorph/X in range(action_def.get_gather_range(), src) - src)
+	var/list/allies_in_range = list()
+	for(var/mob/living/carbon/X in range(action_def.firewalk_range, src))
 		if(X.stat == DEAD)
 			continue
-		if(X.mutation_type == CARRIER_SHAMAN) // Shamans are disconnected from the effect
-			continue
-		if(!hive.is_ally(X))
+		if(!can_not_harm(X))
 			continue
 
-		to_chat(X, SPAN_XENONOTICE("The death of an ovomorph causes you to release adrenal pheromones!"))
+		to_chat(X, SPAN_XENONOTICE("The death of an ovomorph causes you to release fire resistant pheromones!"))
 
 		effect_overlay.flick_overlay(X, 20)
-		effect_power++
+		allies_in_range += X
 
 	addtimer(CALLBACK(BD, /datum/behavior_delegate/carrier_shaman.proc/reset_shaman_ability), action_def.get_cooldown())
 
-	if(BD.enable_pain_usage && effect_power < BD.remembered_count)
-		to_chat(src, SPAN_XENOWARNING("You use stored pain memory."))
-		effect_power = BD.remembered_count
-		BD.remembered_count = 0
-
-	if(!effect_power)
-		to_chat(src, SPAN_XENOWARNING("There weren't enough of your allies around for the sacrifice to be effective."))
-		return
-
-	var/effect = min(((effect_power - 1) * action_def.pheromone_strength_per_xeno) + action_def.pheromone_strength_base, 5)
+	for(var/mob/living/carbon/X as anything in allies_in_range)
+		if(isXenoQueen(X))
+			continue
+		X.ExtinguishMob()
+		X.mob_flags |= FIRE_SHIELD_ON
+		RegisterSignal(X, COMSIG_LIVING_PREIGNITION, .proc/firewalk_immune_on)
+		RegisterSignal(X, list(COMSIG_LIVING_FLAMER_FLAMED,), .proc/firewalk_callback)
```

Comment:
doesn't make them immune to mortar/ob/other fire

Reply:
Should still work for walking over other fires, as far as I'm aware, since it seems to be using Pyro Armor FIREWALK code. Was it visually seen not working during a TM?

Separate concern: Not sure if ability scope is too wide though, since that ability also prevents ignition from direct flames, and the PR description only lists allowing targets to walk over fire + extinguishing active ignitions.